### PR TITLE
Fix: Disable dlna server API responses if dlna is disabled.

### PR DESCRIPTION
**Pull request by BaronGreenback**
Fix for https://github.com/jellyfin/jellyfin/issues/4431

The Dlna server class is instantiated whether or not the dlna server is enabled or not. (the attribute only controls the comms).

This change caused http not found to be returned if the dlna server is disabled.

### CODE REVIEW

**Comment by crobibero** [comment]
If DLNA was staying in core server I would recommend disabling the endpoints using and auth policy, but this seems to be fine.

Don't forget to add to the endpoint docs when adding the `ProducesResponseType`

**Comment by BaronGreenback** [comment]
Wanted to do as little as possible because of that. Testing on the dlna plugins is coming along nicely.

**Comment by crobibero** [comment]
Should probably make the remaining endpoints also return 404

File: Jellyfin.Api/Controllers/DlnaServerController.cs [hunk 1]
```diff
@@ -117,11 +145,17 @@ public ActionResult GetConnectionManager([FromRoute, Required] string serverId)
         /// <returns>Control response.</returns>
         [HttpPost("{serverId}/ContentDirectory/Control")]
         [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/DlnaServerController.cs:148]: Add XML docs

File: Jellyfin.Api/Controllers/DlnaServerController.cs [hunk 2]
```diff
@@ -132,26 +166,39 @@ public async Task<ActionResult<ControlResponse>> ProcessContentDirectoryControlR
         /// <returns>Control response.</returns>
         [HttpPost("{serverId}/ConnectionManager/Control")]
         [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/DlnaServerController.cs:169]: Add XML docs

File: Jellyfin.Api/Controllers/DlnaServerController.cs [hunk 3]
```diff
@@ -165,74 +212,108 @@ public async Task<ActionResult<ControlResponse>> ProcessMediaReceiverRegistrarCo
         [ApiExplorerSettings(IgnoreApi = true)] // Ignore in openapi docs
         [SuppressMessage("Microsoft.Performance", "CA1801:ReviewUnusedParameters", MessageId = "serverId", Justification = "Required for DLNA")]
         [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/DlnaServerController.cs:215]: Add XML docs

**Review by crobibero — APPROVED** [review]

**Review by Ullmie02 — COMMENTED** [review]
I would prefer if the 404 docs were more descriptive like "DLNA disabled"

**Comment by BaronGreenback** [comment]
Could return 503 - Service Unavailable ?

**Comment by crobibero** [comment]
> 
> 
> Could return 503 - Service Unavailable ?

That makes sense to me

File: Jellyfin.Api/Controllers/DlnaServerController.cs [hunk 4]
```diff
@@ -41,198 +41,289 @@ public DlnaServerController(IDlnaManager dlnaManager)
...
-            return Ok(xml);
+            if (DlnaEntryPoint.Enabled)
+            {
+                var url = GetAbsoluteUri();
+                var serverAddress = url.Substring(0, url.IndexOf("/dlna/", StringComparison.OrdinalIgnoreCase));
+                var xml = _dlnaManager.GetServerDescriptionXml(Request.Headers, serverId, serverAddress);
+                return Ok(xml);
+            }
+
+            return StatusCode(503);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/DlnaServerController.cs:62]:
```suggestion
            return StatusCode(StatusCodes.Status503ServiceUnavailable)
```

File: Jellyfin.Api/Controllers/DlnaServerController.cs [hunk 5]
```diff
@@ -41,198 +41,289 @@ public DlnaServerController(IDlnaManager dlnaManager)
         /// </summary>
         /// <param name="serverId">Server UUID.</param>
         /// <response code="200">Description xml returned.</response>
+        /// <response code="503">Service Unavailable.</response>
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/DlnaServerController.cs:44]:
```suggestion
        /// <response code="503">DLNA is disabled.</response>
```

**Review by crobibero — COMMENTED** [review]
repeat for all endpoints

**Review by crobibero — APPROVED** [review]

**Review by Ullmie02 — APPROVED** [review]

**Pull request merged**